src/app/courses/add-courses/courses-add.component.ts

```diff
@@ -126,13 +126,23 @@ export class CoursesAddComponent implements OnInit {
 
   addStep() {
     this.steps.push({
+      id: this.uniqueIdOfStep(),
       stepTitle: '',
       description: ''
     });
   }
 
-  removeStep(pos) {
-    this.steps.splice(pos, 1);
+  uniqueIdOfStep() {
+     // Math.random should be unique because of its seeding algorithm.
+     // Convert it to base 36 (numbers + letters), and grab the first 9 characters
+    // after the decimal.
+    return '_' + Math.random().toString(36).substr(2, 9);
+  }
+
+  removeStep(step) {
```

Comment:
why is this removestep required need to change?

Reply:
it seems like you already added filter which is breaking it instead of removing item from array